pkg/virt-api/rest/subresource.go

```diff
@@ -464,9 +489,60 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 		return
 	}
 
+	// Only force restart with GracePeriodSeconds=0 is supported for now
+	// Here we are deleting the Pod because CRDs don't support gracePeriodSeconds at the moment
+	if bodyStruct.GracePeriodSeconds != nil {
+		if *bodyStruct.GracePeriodSeconds == 0 {
+			vmiPodname, err := app.findPod(namespace, vmi)
+			if err != nil {
+				response.WriteError(http.StatusForbidden, err)
+				return
+			}
```

Comment:
I think that you need something like

```go
if vmiPodname == "" {
  response.WriteHeader(http.StatusAccepted)
  return
}
```

Reply:
I think that the unit tests are not covering this scenaro btw. You will have to create a VMI in the tests which is `Running`, but where the pod is not `Running`. That can for instance happen in real life, until the VMI controller has resynced with the pod state.

Reply:
Added.